### PR TITLE
AMD: increase memory pool

### PR DESCRIPTION
**Pull request by psychocrypt**
Documentation: Add environment variables to increase OpenCL memory pool

### CODE REVIEW

**Comment by fireice-uk** [comment]
Maybe we should just set them from the miner using `setenv`?

**Comment by psychocrypt** [comment]
Mhhh I also thought about it but found no way how I can do it before the
start of the miner. But since we are loading the opencl lib with via shared
library it could be possible to set the environment variables before.
I will test it.

File: doc/tuning.md [hunk 1]
```diff
@@ -66,3 +67,17 @@ If you are unsure of either GPU or platform index value, you can use `clinfo` to
...
 "platform_index" : 0,
 ```
+
+### Increase Memory Pool
+
+By setting the following environment variables before the miner is started OpenCl allows the miner to more threads.
+This variables must be set each time before the miner is started else it could be that the miner can not allocate enough memory and is crashing.
+
+```
+export GPU_FORCE_64BIT_PTR 1
```
Comment (psychocrypt) [inline doc/tuning.md]: `=` is missing

**Comment by psychocrypt** [comment]
@fireice-uk Could you please run the miner with amd auto suggestion once with the PR and once with the dev version.
The difference should be visible in the command over the gpu settings. If it is working than `... memory: XXX` should be larger with this PR.

**Comment by fireice-uk** [comment]
```
fireice@archerg:~/psc-xmr-stak/build/bin$ cat amd.txt

/*
 * GPU configuration. You should play around with intensity and worksize as the fastest settings will vary.
 *      index    - GPU index number usually starts from 0
 *  intensity    - Number of parallel GPU threads (nothing to do with CPU threads)
 *   worksize    - Number of local GPU threads (nothing to do with CPU threads)
 * affine_to_cpu - This will affine the thread to a CPU. This can make a GPU miner play along nicer with a CPU miner.
 * "gpu_threads_conf" :
 * [
 *	{ "index" : 0, "intensity" : 1000, "worksize" : 8, "affine_to_cpu" : false },
 * ],
 */

"gpu_threads_conf" : [
  // gpu: Ellesmere memory:3920
  // compute units: 36
  { "index" : 0,
    "intensity" : 864, "worksize" : 8,
    "affine_to_cpu" : false, 
  },

],

/*
 * Platform index. This will be 0 unless you have different OpenCL platform - eg. AMD and Intel.
 */
"platform_index" : 0,

fireice@archerg:~/psc-xmr-stak/build/bin$ cat amd_.txt

/*
 * GPU configuration. You should play around with intensity and worksize as the fastest settings will vary.
 *      index    - GPU index number usually starts from 0
 *  intensity    - Number of parallel GPU threads (nothing to do with CPU threads)
 *   worksize    - Number of local GPU threads (nothing to do with CPU threads)
 * affine_to_cpu - This will affine the thread to a CPU. This can make a GPU miner play along nicer with a CPU miner.
 * "gpu_threads_conf" :
 * [
 *	{ "index" : 0, "intensity" : 1000, "worksize" : 8, "affine_to_cpu" : false },
 * ],
 */

"gpu_threads_conf" : [
  // gpu: Ellesmere memory:3920
  // compute units: 36
  { "index" : 0,
    "intensity" : 864, "worksize" : 8,
    "affine_to_cpu" : false, 
  },

],

/*
 * Platform index. This will be 0 unless you have different OpenCL platform - eg. AMD and Intel.
 */
"platform_index" : 0,
```
Note, my RX480 has only 4GB of memory - do you need more?

**Comment by psychocrypt** [comment]
@fireice-uk Thanks for testing. It could be that the limitation is removed within newer drivers. If @anhphan can not test it until Wednesday than I will remove `setenv` from the PR and keep only the possible tuning option in the documentation.

@anhphan Could you please test this PR if the auto suggestion is setting the intensity to 2000 if you are using this PR.

**Comment by psychocrypt** [comment]
@fireice-uk I forced push a change, the environment variable will be still set but only if the use not setting something else in his environment before.
This means that this PR has no negative side effect for the user but possible increase the performance.

**Comment by anhphan** [comment]
@psychocrypt Just have test this, delete all user variable set by `setx` command before. Not sure what need to check but auto suggestion for amd not use 2000 intensity:

```

/*
 * GPU configuration. You should play around with intensity and worksize as the fastest settings will vary.
 *      index    - GPU index number usually starts from 0
 *  intensity    - Number of parallel GPU threads (nothing to do with CPU threads)
 *   worksize    - Number of local GPU threads (nothing to do with CPU threads)
 * affine_to_cpu - This will affine the thread to a CPU. This can make a GPU miner play along nicer with a CPU miner.
 * "gpu_threads_conf" :
 * [
 *	{ "index" : 0, "intensity" : 1000, "worksize" : 8, "affine_to_cpu" : false },
 * ],
 */

"gpu_threads_conf" : [
  // gpu: gfx901 memory:3920
  // compute units: 64
  { "index" : 0,
    "intensity" : 1536, "worksize" : 8,
    "affine_to_cpu" : false, 
  },
  // gpu: gfx901 memory:3920
  // compute units: 64
  { "index" : 1,
    "intensity" : 1536, "worksize" : 8,
    "affine_to_cpu" : false, 
  },

],

/*
 * Platform index. This will be 0 unless you have different OpenCL platform - eg. AMD and Intel.
 */
"platform_index" : 0,

```

**Comment by psychocrypt** [comment]
@anhphan Thank you for the test, it shows that is is not possible to set the environment variable within the code:-(

@fireice-uk I removed the second commit where the environment variable is set by the miner itself. This means This PR will only extent the documentation.

**Pull request merged**